### PR TITLE
modbus: bump up rust crate version

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
cf https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=47003

Describe changes:
- bump up modbus crate version

cc @cccs-sadugas is there a good way to get modbus parser as a once inited constant ?

cf https://github.com/CybercentreCanada/sawp/pull/1

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on smb_collect.
  

Pipeline 7515

File: rust/src/modbus/modbus.rs [hunk 1]
```diff
@@ -183,8 +182,9 @@ impl ModbusState {
 
     pub fn parse(&mut self, input: &[u8], direction: Direction) -> AppLayerResult {
         let mut rest = input;
+        let modbus_parser = sawp_modbus::Modbus::default();
```
Comment (cccs-sadugas) [inline rust/src/modbus/modbus.rs:185]: The `probe_strict` option is off by default and I think you intend to enable it. However, looking at [the source](https://docs.rs/sawp-modbus/0.10.0/src/sawp_modbus/lib.rs.html#471), I forgot to make the field public. I'll release a patch for that.

**Comment by cccs-sadugas** [comment]
> cc @cccs-sadugas is there a good way to get modbus parser as a once inited constant ?

This could be related to my other [comment](https://github.com/OISF/suricata/pull/7421#discussion_r880488400). I'll fix as it should be possible to init the parser once as a const.

**Comment by cccs-sadugas** [comment]
[sawp-0.11.0](https://github.com/CybercentreCanada/sawp/releases/tag/sawp-0.11.0) is now released so it should work as a const like so https://github.com/cccs-sadugas/suricata/commit/dcfbfa2a984f700e0284716f4d15aaebffaf132c

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7442

**Pull request closed without merging**